### DISCUSSION TITLE
Adding/exporting environment variables in bulk

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [craftcms/cloud#38] [Feature Requests & Enhancements]
Currently you must add environment variables per line in a matrix field, but you can add at least add multiple in one go.

It would be great if you could add in bulk in as a single textarea field:

```
VAR1=value
VAR2=value
VAR3="a value that needs quotes"
```

Which then gets processed as key=value per line.

In addition for copying an environments variables in bulk i.e. copying down to a dev environment, it would be nice to be able to export the current variables from Craft Cloud into a .env file.

**Comment by timkelty** [comment]
> It would be great if you could add in bulk in as a single textarea field:

@jamesmacwhite we _almost_ have that feature. You can drag an .env file in and it will parse all the values that way.

![CleanShot 2024-10-25 at 20 13 16@2x](https://github.com/user-attachments/assets/46238461-3aea-4bda-8da9-ed2ad021c19c)

Agreed exporting to an .env file would be nice!

**Reply by jamesmacwhite** [reply]
@timkelty Ah, I guess you can do the importing of, but not the exporting. Exporting would be great, use case, provisioning a local environment from Craft Cloud environment with environment variables populated.

**Comment by timkelty** [comment]
This is now implemented!
